adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -1,47 +1,27 @@
 package caliban
 
-import caliban.execution.QueryExecution
 import caliban.interop.tapir.ws.Protocol
-import caliban.interop.tapir.{ RequestInterceptor, TapirAdapter, WebSocketHooks }
+import caliban.interop.tapir.{ HttpAdapter, WebSocketHooks }
 import sttp.tapir.Codec.JsonCodec
 import sttp.tapir.DecodeResult
 import sttp.tapir.server.ziohttp.{ ZioHttpInterpreter, ZioHttpServerOptions }
-import zio.http._
+import zio._
 import zio.http.ChannelEvent.UserEvent.HandshakeComplete
 import zio.http.ChannelEvent.{ ChannelRead, UserEventTriggered }
+import zio.http._
 import zio.http.socket._
-import zio._
 import zio.stream._
 
 object ZHttpAdapter {
 
-  def makeHttpService[R, E](
-    interpreter: GraphQLInterpreter[R, E],
-    skipValidation: Boolean = false,
-    enableIntrospection: Boolean = true,
-    queryExecution: QueryExecution = QueryExecution.Parallel,
-    requestInterceptor: RequestInterceptor[R] = RequestInterceptor.empty
-  )(implicit
-    requestCodec: JsonCodec[GraphQLRequest],
-    responseCodec: JsonCodec[GraphQLResponse[E]],
+  def makeHttpService[R, E](adapter: HttpAdapter[R, E])(implicit
     serverOptions: ZioHttpServerOptions[R] = ZioHttpServerOptions.default[R]
-  ): HttpApp[R, Throwable] = {
-    val endpoints = TapirAdapter.makeHttpService[R, E](
-      interpreter,
-      skipValidation,
-      enableIntrospection,
-      queryExecution,
-      requestInterceptor
-    )
-    ZioHttpInterpreter(serverOptions).toHttp(endpoints)
-  }
+  ): HttpApp[R, Throwable] =
+    ZioHttpInterpreter(serverOptions).toHttp(adapter.serverEndpoints[R])
 
   def makeWebSocketService[R, E](
     interpreter: GraphQLInterpreter[R, E],
-    skipValidation: Boolean = false,
```

Comment:
Argh, the zio-http websocket support still doesn't use Tapir! Maybe it's doable though.

Reply:
Solved